### PR TITLE
Add DNS support to loom_base

### PR DESCRIPTION
**Pull request by wolf31o2**
Currently, we only support dnsimple, but any DNS provider which can be modified via API can be added. By default, this does nothing, so it's safe to add everywhere. To use this, a user would need to add JSON to his `base` service in Loom such as:

``` json
{"loom_dns":{"provider":"dnsimple"}}
```

This would require the use of a `creds` data bag with a `dnsimple` item. These are configurable, and someone can choose to provide an username/password combo, instead.

``` json
{"loom_dns":{"provider":"dnsimple","dnsimple":{"use_databag":false}},"dnsimple":{"username":USERNAME,"password":PASSWORD}}
```

### CODE REVIEW

File: provisioner/worker/plugins/automators/chef_solo_automator/chef_solo_automator/cookbooks/loom_dns/metadata.rb [hunk 1]
```diff
@@ -0,0 +1,10 @@
+name             'loom_dns'
+maintainer       'Cask'
```
Comment (dereklwood) [inline provisioner/worker/plugins/automators/chef_solo_automator/chef_solo_automator/cookbooks/loom_dns/metadata.rb]: fyi... doesn't match the copyright headers

Comment (wolf31o2) [inline provisioner/worker/plugins/automators/chef_solo_automator/chef_solo_automator/cookbooks/loom_dns/metadata.rb]: Fixed.

File: provisioner/worker/plugins/automators/chef_solo_automator/chef_solo_automator/cookbooks/loom_dns/recipes/dnsimple.rb [hunk 2]
```diff
@@ -0,0 +1,55 @@
...
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+#
+
+%w(build-essential dnsimple).each do |recipe|
+  include_recipe recipe
+end
+
+if node['loom_dns']['dnsimple']['use_databag']
```
Comment (dereklwood) [inline provisioner/worker/plugins/automators/chef_solo_automator/chef_solo_automator/cookbooks/loom_dns/recipes/dnsimple.rb]: the `use_databag` attribute seems unecessary and burdensome to have to set to false if you want to use the dnsimple username/password.  why dont we just use `node['loom_dns']['dnsimple']...` if defined, else default to `node['dnsimple']...`, else fail ?

**Comment by dereklwood** [comment]
minor comment otherwise :+1: lgtm

File: provisioner/worker/plugins/automators/chef_solo_automator/chef_solo_automator/cookbooks/loom_dns/recipes/dnsimple.rb (hunk 2 above)
Comment (wolf31o2) [inline provisioner/worker/plugins/automators/chef_solo_automator/chef_solo_automator/cookbooks/loom_dns/recipes/dnsimple.rb]: Why `node['loom_dns']['dnsimple']` and not just a check directly on `node['dnsimple']`?

**Pull request merged**